ckanext/unaids/tests/test_plugin.py

```diff
@@ -42,52 +42,85 @@ class TestResourceLastModified(object):
     '''
 
-    URL_TYPE_FILE_UPLOAD = 'upload'
-    URL_TYPE_LINK = ''
-
-    def test_resource(self, url_type):
-        resource_dict = {'url_type': url_type}
-        if url_type == self.URL_TYPE_FILE_UPLOAD:
-            resource_dict.update({
-                'url': 'file.csv',
-                'lfs_prefix': 'prefix123',
-                'sha256': 'sha256',
-                'size': 100,
-            })
-        elif url_type == self.URL_TYPE_LINK:
-            resource_dict.update({
-                'url': 'http://example.com'
-            })
-        return resource_dict
-
-    def test_null_to_file(self):
-        resource = self.test_resource(self.URL_TYPE_FILE_UPLOAD)
-        _update_resource_last_modified_date(resource=resource)
-        assert 'last_modified' in resource
-
-    def test_file_to_null(self):
-        resource = self.test_resource(self.URL_TYPE_FILE_UPLOAD)
-        _update_resource_last_modified_date({}, resource)
-        assert 'last_modified' not in resource
-
-    def test_file_to_url(self):
-        current = self.test_resource(self.URL_TYPE_FILE_UPLOAD)
-        resource = self.test_resource(self.URL_TYPE_LINK)
-        _update_resource_last_modified_date(resource, current)
-        assert 'last_modified' in resource
-
-    def test_null_to_url(self):
-        resource = self.test_resource(self.URL_TYPE_LINK)
-        _update_resource_last_modified_date(resource=resource)
-        assert 'last_modified' in resource
-
-    def test_url_to_null(self):
-        resource = self.test_resource(self.URL_TYPE_LINK)
-        _update_resource_last_modified_date({}, resource)
-        assert 'last_modified' not in resource
-
-    def test_url_to_file(self):
-        current = self.test_resource(self.URL_TYPE_LINK)
-        resource = self.test_resource(self.URL_TYPE_FILE_UPLOAD)
-        _update_resource_last_modified_date(resource, current)
-        assert 'last_modified' in resource
+    @pytest.fixture
+    def resource_with_file(autouse=True):
+        return {
+            'url_type': 'upload',
+            'url': 'file.csv',
+            'lfs_prefix': 'prefix123',
+            'sha256': 'sha256',
+            'size': 100,
+        }
+
+    @pytest.fixture
+    def resource_with_link(autouse=True):
+        return {
+            'url_type': '',
+            'url': 'http://example.com'
+        }
+
+    def test_null_to_file(self, resource_with_file):
+        _update_resource_last_modified_date(resource_with_file)
+        assert 'last_modified' in resource_with_file
+
+    def test_file_to_null(self, resource_with_file):
+        null_resource = {}
+        _update_resource_last_modified_date(
+            null_resource, current=resource_with_file
+        )
+        assert 'last_modified' not in resource_with_file
+
+    def test_file_to_file(self, resource_with_file):
+        updated_resource = resource_with_file.copy()
+        updated_resource.update({'url': 'file2.csv'})
```

Comment:
Simply create another fixture
```
@fixture
def updated_resource_with_file():
 # returning dict with new 'url' value
```

Reply:
I'd rather not as the fixtures would only be used once so there's no need to take it out into it's own function. I also want to be super clear that i'm taking the original dictionary and only updating a single field, rather than the entire thing.

Reply:
Making it another fixture will look just so much better. And it'll be easier to reuse the test input in the future.
If you really want to keep the intent of overwriting just a single field you can use:
```
@fixture
def original():
    return {'url': 'foo', 'description': 'bar'}

@fixture
def updated(original):
     return original.copy().update({'description': 'baz'})
```